### PR TITLE
Adding T_SolAir Fluxes

### PR DESCRIPTION
**Pull request by lorenzogracia**
- Adding new process for computing heat fluxes accoding t sol air theory
- Including FACE_HEAT_FLUX in thermal solver
- Updating check and prepare model accoding new thermal loads
- Updating test_examples involved in thermal problem

### CODE REVIEW

File: applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    void ExecuteInitializeSolutionStep()
+    {
+        
+        KRATOS_TRY;
+
+        const int nnodes = mrModelPart.GetMesh(mMeshId).Nodes().size();
+        Variable<double> var = KratosComponents< Variable<double> >::Get(mVariableName);
+        
+        KRATOS_WATCH(mTableId)
```
Comment (rubenzorrilla) [inline applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp]: Missing KRATOS_WATCH

File: applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        {
+            ModelPart::NodesContainerType::iterator it_begin = mrModelPart.GetMesh(mMeshId).NodesBegin();
+
+            #pragma omp parallel for
+            for(int i = 0; i<nnodes; i++)
+            {
+                ModelPart::NodesContainerType::iterator it = it_begin + i;
+
+                double temp_current = it->FastGetSolutionStepValue(TEMPERATURE);
+                double heat_flux = mH0*(t_sol_air - temp_current);
```
Comment (rubenzorrilla) [inline applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp]: These two variables can be const.

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]

**Comment by lorenzogracia** [comment]
Hi Ruben, thank you for your comments. I have done a commit with the requested changes.

**Review by ipouplana — APPROVED** [review]

**Review by rubenzorrilla — APPROVED** [review]

File: applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            {
+                "model_part_name":"PLEASE_CHOOSE_MODEL_PART_NAME",
+                "mesh_id": 0,
+                "variable_name": "PLEASE_PRESCRIBE_VARIABLE_NAME",
+                "h_0"                             : 0.0,
+                "ambient_temperature"             : 0.0,
+                "table_ambient_temperature"       : 0,
+                "emisivity"                       : 0.0,
+                "delta_R"                         : 0.0,
+                "absortion_index"                 : 0.0,
```
Comment (pooyan-dadvand) [inline applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp:59]: @lorenzogracia I think the absorption is misspelled here...

File: applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+    /// Member Variables
+    ModelPart& mrModelPart;
+    std::size_t mMeshId;
+    std::string mVariableName;
+    double mH0;
+    double mAmbientTemperature;
+    double mEmisivity;
+    double mDeltaR;
+    double mAbsortion_index;
```
Comment (pooyan-dadvand) [inline applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp:198]: Here Also

Comment (lorenzogracia) [reply applications/DamApplication/custom_processes/dam_t_sol_air_heat_flux_process.hpp:198]: Thanks @pooyan-dadvand, I will change them.

**Pull request merged**